YAFC/Utils/CommandLineParser.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.IO;
+using System;
+
+namespace YAFC {
+    public static class CommandLineParser {
+        public static string lastError { get; private set; } = string.Empty;
+        public static bool helpRequested { get; private set; }
+
+        public static bool errorOccured => !string.IsNullOrEmpty(lastError);
+
+        public static ProjectDefinition Parse(string[] args) {
+            ProjectDefinition options = new ProjectDefinition();
+
+            if (args.Length == 0) {
+                return options;
+            }
+
+            if (!args[0].StartsWith("--")) {
+                options.dataPath = args[0];
```

Comment:
I think it would be better to first validate the value and then to assign it because in this case we don't assign incorrect data to the `options`. The PR's implementation has a potential for trouble if the inserted data is automatically propagated somewhere else in the future. That applies to all parameters.
The pseudocode is something like
```
if (!validationCondition) {
report error;
return null;
}
assign option property;
```

Reply:
Right now, the project object is directly used in `WelcomeScreen.cs`, which has a validator with `WelcomeScreen.ValidateSelection()`.
So, validating in the `CommandLineParser` would lead to redundant code. Since `WelcomeScreen` has different ways to populate a `ProjectDefinition`, we can't simply move the validation code into `CommandLineParser`.

But what we can do is to move all the validation code into `ProjectDefinition`, since that would adhere better to Single Responsibility. From there, `CommandLineParser`, `WelcomeScreen`, and any other class could use it without a hassle.

What do you think?

Reply:
Addendum:
People keep experiencing crashes and we strongly suspect that it's because of `ProjectDefinition`s that are `null`, don't we?
So I'd like to get this PR on the road to mitigate this.
I'd submit a followup PR that refactors the validation, if needed/desired.

Reply:
> People keep experiencing crashes

If you're talking about the crashes mentioned [there](https://github.com/have-fun-was-taken/yafc-ce/issues/82), it's because the [fix](https://github.com/have-fun-was-taken/yafc-ce/pull/93) was not in the prevoius release. I've made [RC2](https://github.com/have-fun-was-taken/yafc-ce/releases/tag/0.6.3-RC2) yesterday that includes it.

Reply:
> move all the validation code into ProjectDefinition

Good idea! I'll summarize it to see if I got it correctly. 

`CommandLineParser` is responsible for parsing the CLI arguments and returning a `ProjectDefinition` instance.  
The class `ProjectDefinition` checks the correctness of the input data in its own constructor.

Reply:
Exactly. Plus the possibility to create other `ProjectDefinition`s (e.g. in the list of recent projects). Regardless of the source, the objects then can validate themselves.

Documentation is a good hint, I tend to forget it 😉